File: src/main/java/com/group/InternMap/Application/ApplicationStatus.java
package com.group.InternMap.Application;

import java.io.Serializable;

public enum ApplicationStatus implements Serializable {

    ACCEPTED(1),
    REJECTED(2),
    PENDING(3);

    private final int id;

    ApplicationStatus(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

}

File: Frontend/app/Model/ApplicationStatus.ts
export enum ApplicationStatus {
    ACCEPTED = "ACCEPTED",
    REJECTED = "REJECTED",
    PENDING = "PENDING",
}

export const ApplicationStatusId: Record<ApplicationStatus, number> = {
    [ApplicationStatus.ACCEPTED]: 1,
    [ApplicationStatus.REJECTED]: 2,
    [ApplicationStatus.PENDING]: 3,
};